### PR TITLE
Feature: Allow the admin to overwrite the modules purpose, solves #288.

### PR DESCRIPTION
**Pull request by abias**
This PR is a DevCamp result at MoodleMootDACH with very warm thanks to [Academic Moodle Cooperation](https://moodle.academic-moodle-cooperation.org/) for providing existing code from their their Boost child theme.

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -280,6 +285,41 @@ function theme_boost_union_get_extra_scss($theme) {
...
+        }
+        // If the activity purpose setting is set and differs from the activity's default purpose.
+        $configname = 'activitypurpose'.$modname;
+        if (isset($theme->settings->{$configname}) && $theme->settings->{$configname} != $defaultpurpose) {
+            // Add CSS to modify the activity purpose color in the activity chooser and the activity icon.
+            $content .= '.activity.modtype_'.$modname.' .activityiconcontainer.courseicon,';
+            $content .= '.modchoosercontainer .modicon_'.$modname.'.activityiconcontainer { ';
+            $content .= 'background-color: var(--activity'.$theme->settings->{$configname}.') !important;';
+            $content .= '}';
+            // If the default purpose was 'other', make the icon white.
```
Comment (lucaboesch) [inline lib.php:313]: I didn't understand the comment and propose to extend it.
`// If the default purpose originally was 'other' and now is overridden, make the icon white.`

File: settings.php [hunk 2]
```diff
@@ -389,6 +390,44 @@
         $setting->set_updatedcallback('theme_reset_all_caches');
         $tab->add($setting);
 
+        // Create activity icons purpose heading.
+        $name = 'theme_boost_union_amc/activitypurposeheading';
```
Comment (lucaboesch) [inline settings.php:394]: remove AMC

**Review by lucaboesch — CHANGES_REQUESTED** [review]
Those small changes, and then you're good to go.

**Comment by abias** [comment]
Thank you, Luca. Both issues have been fixed now.

Cheers,
Alex

**Pull request merged**